lib/gooddata/lcm/actions/synchronize_clients.rb

```diff
@@ -98,6 +114,36 @@ def call(params)
           # Return results
           results
         end
+
+        def remove_old_master_workspace(params, segment_id, master_projects, number_of_deleted_projects)
+          removal_master_project_ids = []
+          latest_project = master_projects.pop
+          removal_projects = master_projects.shift(number_of_deleted_projects)
+          # rubocop:disable Style/RescueStandardError
+          removal_projects.each do |project_wrapper|
+            project = params.gdc_gd_client.projects(project_wrapper[:master_project_id])
+            begin
+              GoodData.logger.info "Segment #{segment_id}: Deleting old master workspace, project: '#{project.title}', PID: (#{project.pid})."
+              project.delete if project && !project.deleted?
+              removal_master_project_ids << project_wrapper[:master_project_id]
+            rescue => ex
+              GoodData.logger.error "Unable to remove master workspace, reason: #{ex}"
```

Comment:
"Unable to remove master workspace: '#{master_project_id}', Error: #{ex}"

Reply:
updated it